pkg/plugins/policies/meshloadbalancingstrategy/api/v1alpha1/validator.go

```diff
@@ -49,6 +50,88 @@ func validateTo(to []To) validators.ValidationError {
 func validateDefault(conf Conf) validators.ValidationError {
 	var verr validators.ValidationError
 	verr.AddError("loadBalancer", validateLoadBalancer(conf.LoadBalancer))
+	verr.AddError("localityAwareness", validateLocalityAwareness(conf.LocalityAwareness))
+	return verr
+}
+
+func validateLocalityAwareness(localityAwareness *LocalityAwareness) validators.ValidationError {
+	var verr validators.ValidationError
+	if localityAwareness == nil {
+		return verr
+	}
+	verr.AddError("localZone", validateLocalZone(localityAwareness.LocalZone))
+	verr.AddError("crossZone", validateCrossZone(localityAwareness.CrossZone))
+	verr.AddError("failoverThreshold", validateFailoverThreshold(localityAwareness.FailoverThreshold))
+	return verr
+}
+
+func validateLocalZone(localZone *LocalZone) validators.ValidationError {
+	var verr validators.ValidationError
+	if localZone == nil {
+		return verr
+	}
+
+	var weightSpecified int
+	for idx, affinityTag := range localZone.AffinityTags {
+		path := validators.RootedAt("affinityTags").Index(idx)
+		if affinityTag.Weight != nil {
+			verr.Add(validators.ValidateIntegerGreaterThan(path.Field("weight"), *affinityTag.Weight, 1))
+			weightSpecified++
+		}
+	}
+
+	if weightSpecified > 0 && weightSpecified != len(localZone.AffinityTags) {
+		verr.AddViolation("affinityTags", "each or none affinity tags should have weight")
+	}
+	return verr
+}
+
+func validateCrossZone(crossZone *CrossZone) validators.ValidationError {
+	var verr validators.ValidationError
+	if crossZone == nil {
+		return verr
+	}
+
+	for idx, failover := range crossZone.Failover {
+		path := validators.RootedAt("failover").Index(idx)
+		if failover.From != nil {
+			if len(failover.From.Zones) == 0 {
+				verr.AddViolationAt(path.Field("from").Field("zones"), "must not be empty")
```

Comment:
should probably use https://github.com/kumahq/kuma/blob/95ab5c494973eb62d74184c75f4ee21c8f2ff569/pkg/core/validators/messages.go#L11